client/admin_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package client_test
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cerbos/cerbos/client"
+	"github.com/cerbos/cerbos/client/testutil"
+	"github.com/cerbos/cerbos/internal/audit"
+	"github.com/cerbos/cerbos/internal/audit/local"
+)
+
+func TestAuditLogs(t *testing.T) {
```

Comment:
So, this test is not really doing what you expect :-). It's still reading from the `NopLog` which never errors. Even if you remove all the data loading calls, the test will still pass. The returned data is always `nil`.

I didn't expect the actual audit log to be ever used in tests to store and retrieve data. So the real audit log tries to read its configuration when you call `NewLog`. In the test environment, there's no configuration file and it falls back to `NopLog` when that's the case.  Therefore, the `WithAudit` option that has been added here does not really do anything. 

If we really want to be able to store and retrieve data from the audit log during tests, then we'll have to refactor the audit package to not rely on the config file being present and introduce other test-friendly options like using the Badger in-mem mode and removing the flush delay etc. 

If it were me, I would just find a way to test the _new_ code added here to make sure it works. The audit log has its own tests so we can be fairly confident that the interface we use to communicate with it is going to work as expected.

Reply:
I see, the best option is to add a audit log mock to the test server 👍